### ISSUE TITLE
Reducing time taken to predict probabilities 

### ISSUE THREAD

**Issue by guptshiva** [guptshiva/CodeBlooded#1]
Hi 

I am using Keras with Tensorflow backend. Once I have the model I am using model.predict_proba to predict probabilities. The time taken by this predict function seems to be independent of the size of the array, or rather less dependent on the size of the array that it is predicting. 

What can I do to reduce this time taken? 

First I run the loop 100 times and time the predictions as below 

fname_model = folder+"/my_modelbuy_rocbased.h5"
best_model = load_model(fname_model)
best_model.compile(loss='categorical_crossentropy', optimizer='Adam', metrics=['acc'])
X_test_orig = pd.read_csv(folder+"/ret_out_20170402.csv",sep=",")
x_test = X_test_orig.as_matrix()
y1=np.array([x_test[1]])
print y1.shape 
t1=datetime.now().strftime("%H:%M:%S.%f")
for i in range(0,100,1):
	prob1= best_model.predict_proba(y1)
t2=datetime.now().strftime("%H:%M:%S.%f")	
print "time taken ", t1,t2

**********************************************After Running***********************

time taken  11:50:53.909226 11:50:54.372596

____________________________________________________________________________________

So for 100 predictions, the time taken was 11:50:54.372596 minus 11:50:53.909226 which is about 463 milliseconds, which would imply 4.63 milliseconds per prediction 

However if i run the loop for just one prediction as shown below , 

fname_model = folder+"/my_modelbuy_rocbased.h5"
best_model = load_model(fname_model)
best_model.compile(loss='categorical_crossentropy', optimizer='Adam', metrics=['acc'])
X_test_orig = pd.read_csv(folder+"/ret_out_20170402.csv",sep=",")
x_test = X_test_orig.as_matrix()
y1=np.array([x_test[1]])
print y1.shape 
t1=datetime.now().strftime("%H:%M:%S.%f")
for i in range(0,1,1):
	prob1= best_model.predict_proba(y1)
t2=datetime.now().strftime("%H:%M:%S.%f")	
print "time taken ", t1,t2

**********************************************After Running***********************

time taken  11:54:40.683225 11:54:41.144969

____________________________________________________________________________________

So for 1 prediction, the time taken is 11:54:40.683225 minus 11:54:41.144969 which is also about 440 milliseconds. 

There seems to be a fixed amount of time the model will take regardless of the number of calls. How can I reduce this? 

*********************************************What is it that I want to acchieve********************************

In reality, my y1 will always be of shape(1,31). That is I will always only have one sample to predict. Each time it will take me about 400 milliseconds. However, there is some setup that is happening at the backend which may not be required which is why when I for loop through the prediction the time took remains the same . is there a way by which I can get rid of this "additional setup ". Can I keep the graph loaded in the memory so that predictions happen faster?
